packages/headless/src/features/insight-search/insight-search-actions-thunk-processor.test.ts

```diff
@@ -164,4 +164,107 @@ describe('AsyncInsightSearchThunkProcessor', () => {
     });
     expect(processed.automaticallyCorrected).toBe(true);
   });
+
+  describe('query correction processing', () => {
+    describe('legacy query correction processing', () => {
+      it('should correct the query the legacy way', async () => {
+        const processor = new AsyncInsightSearchThunkProcessor<{}>({
+          ...config,
+          getState: vi.fn().mockReturnValue({
+            configuration: getConfigurationInitialState(),
+            search: buildMockSearchState({
+              results,
+              response: buildMockSearchResponse({results}),
+            }),
+            didYouMean: {
+              enableDidYouMean: true,
+              automaticallyCorrectQuery: false,
+            },
+          }),
+        });
+
+        const mappedRequest: MappedSearchRequest<InsightQueryRequest> = {
+          request: buildMockInsightQueryRequest(),
+          mappings: initialSearchMappings(),
+        };
+
+        const searchResponse = buildMockSearchResponse({
+          results: [],
+          queryCorrections: [
+            {
+              correctedQuery: 'bar',
+              wordCorrections: [
+                {
+                  correctedWord: 'foo',
+                  length: 3,
+                  offset: 0,
+                  originalWord: 'foo',
+                },
+              ],
+            },
+          ],
+        });
+
+        const fetched = {
+          response: {
+            success: searchResponse,
+          },
+          duration: 123,
+          queryExecuted: 'foo',
+          requestExecuted: mappedRequest.request,
+        };
+
+        const processed = (await processor.process(
+          fetched
+        )) as ExecuteSearchThunkReturn;
+
+        expect(processed.response).toMatchObject(searchResponse);
+        expect(processed.automaticallyCorrected).toBe(false);
+        expect(processed.queryExecuted).toBe('foo');
+        expect(config.extra.apiClient.query).not.toHaveBeenCalled();
```

Comment:
the test is named `should correct the query` and in these assertion I see the it didn't correct the query :suspect:

Reply:
changed this test